packages/custom_lint/test/cli_test.dart

```diff
@@ -327,4 +328,43 @@ void main() {
       currentDirectory: app,
     );
   });
+
+  test('Correctly exits with errors in ServerToClientChannel.init', () async {
+    final plugin = createPlugin(name: 'test_lint', main: emptyPluginSource);
+
+    final app = createLintUsage(
+      source: {
+        'lib/main.dart': 'void fn() {}',
+        'lib/another.dart': 'void fail() {}',
+      },
+      plugins: {'test_lint': plugin.uri},
+      name: 'test_app',
+    );
+
+    // create error during initialization
+    final packageConfig =
+        File(p.join(app.path, '.dart_tool', 'package_config.json'))
+          ..deleteSync();
+
+    await runWithIOOverride(
+      (out, err) async {
+        await cli.entrypoint();
```

Comment:
Could you use `Process.run` instead of `cli.entrypoint` to achieve this?
This would test that the CLI correctly exits.

Technically this test doesn't guarantee that the command indeed stops; only that it sets the exitCode and logs

Reply:
You wouldn't need runWithIOOverride either then

Reply:
We'd also want a similar test, but for when there's a dependency conflict error.

Reply:
Yea I just tried this but this immediately fails with dependency conflicts on `vm_service` which is constrained to `<10.0.0` in `custom_lint_builder` via `hotreloader`. Not exactly sure how to solve this, probably need to add constraints to the generated pusbspec.yaml

Reply:
What does it conflict with?

Reply:
> We'd also want a similar test, but for when there's a dependency conflict error.

I added a new integration test for this as well but I don't really like the way the conflict is produced. I don't see another way tho.